lib/ui/profile/Profile.dart

```diff
@@ -348,4 +438,196 @@ class Profile extends StatelessWidget {
+  }
+
+  void _inviteFriends() async {
+    await Share.share(_LocalisedStrings.shareText());
+  }
+
+  void _openPrivacyPolicy(BuildContext context) {
+    _navigateToWebView(context, _Strings.privacyPolicyUrl);
+  }
+
+  void _openTermsOfUse(BuildContext context) {
+    _navigateToWebView(context, _Strings.termsOfUseUrl);
+  }
+
+  List<FarmDetailsListItemViewModel> _mapToFarmItemViewModel(
+      Map<String, String> farmDetails) {
+    return farmDetails.entries.map(
+      (MapEntry mapEntry) {
+        return FarmDetailsListItemViewModel(
+          title: mapEntry.key,
+          detail: mapEntry.value,
+        );
+      },
+    ).toList();
+  }
+
+  void _navigateBack(BuildContext context) => Navigator.of(context).pop();
+
+  void _navigateToWebView(BuildContext context, String url) {
+    Navigator.of(context).push(
+      MaterialPageRoute(
+        builder: (context) => WebView(
+          url: url,
+        ),
+      ),
+    );
+  }
+
+  ActionSheet _languageMenu(ProfileViewModel viewModel) {
+    final actions = [
+      ActionSheetListItemViewModel(
+        title: _Strings.englishAction,
+        type: ActionType.selectable,
+        icon: _Icons.englishIcon,
+        checkBoxIcon: _Icons.checkBoxIcon,
+        onTap: () => viewModel.switchLanguageTapped(_Languages.english),
+      ),
+      ActionSheetListItemViewModel(
+        title: _Strings.swahiliAction,
+        type: ActionType.selectable,
+        icon: _Icons.swahiliIcon,
+        checkBoxIcon: _Icons.checkBoxIcon,
+        onTap: () => viewModel.switchLanguageTapped(_Languages.swahili),
+      ),
+    ];
+
+    final actionSheetViewModel = ActionSheetViewModel(
+      actions,
+      _LocalisedStrings.cancel(),
+      confirmButtonTitle: _LocalisedStrings.confirm(),
+    );
+    return ActionSheet(
+      viewModel: actionSheetViewModel,
+      style: ActionSheetStyle.defaultStyle(),
+    );
+  }
+
+  ActionSheet _imagePickerMenu(ProfileViewModel viewModel) {
+    final actions = [
+      ActionSheetListItemViewModel(
+        title: _LocalisedStrings.takePictureFromCamera(),
+        type: ActionType.simple,
+        icon: null,
+        onTap: () => _pickImage(ImagePickerLib.ImageSource.camera, viewModel),
+      ),
+      ActionSheetListItemViewModel(
+        title: _LocalisedStrings.pickImageFromGallery(),
+        type: ActionType.simple,
+        icon: null,
+        onTap: () => _pickImage(ImagePickerLib.ImageSource.gallery, viewModel),
+      ),
+    ];
+
+    final actionSheetViewModel = ActionSheetViewModel(
+      actions, _LocalisedStrings.cancel(),
+    );
+    return ActionSheet(
+      viewModel: actionSheetViewModel,
+      style: ActionSheetStyle.defaultStyle(),
+    );
+  }
+
+  _pickImage(ImagePickerLib.ImageSource imageSource, ProfileViewModel viewModel) {
+    ImagePicker.pickImage(
+      onSuccess: (file) {
+        viewModel.saveProfileImage(file);
+      },
+      onError: (message) {
+        //ignore
```

Comment:
Should be really ignored?

Reply:
Well.. it's not an error it self it's when the user cancel the camera or the gallery so we don't want to do anything

Reply:
But in case it raises not due user interaction, or it will never happen?

Reply:
it will never happen, the onError callback is called when the user cancel the camera, the gallery or the crop. I will change it to onCancel to avoid confusions

Reply:
done